### DISCUSSION TITLE
Border Radius Fix

### DISCUSSION THREAD

**Discussion by lucasartsfr** [rpearce/react-medium-image-zoom#395] [Ideas]
Hello,

data-rmiz-modal-content has an `<img>` with CSS transform property (example: `transform: translate(0px, 0px) scale(0.374583);`)

When applying the same border-radius of the original image to the image that will be zoomed, the result is not good. Indeed, given that the transform Scale is not at 1, the radius is not good, and there is a transition that is not smooth. It would be necessary to divide the border-radius value by the scale value of the image to have the same render and a smooth transition.

Have you ever had this problem and/or a solution to solve it?

Thansk Again for this amazing module !

**Comment by rpearce** [comment]
@lucasartsfr Can you 🙏 provide a simple example of this for me to play with so I can better understand?

**Reply by lucasartsfr** [reply]
Sure, here is a video (can't upload project for the moment, I am sorry)

Both Images have the same Border Radius Value (50px for the test). But because of Scale value, the 50px radius does not look the same, we need to divide it by Scale Value 

https://user-images.githubusercontent.com/26819739/214600942-fb1c87b9-652c-409a-a364-ebe5de49ba81.mp4

![image](https://user-images.githubusercontent.com/26819739/214601533-87e5ef47-14b5-4315-be3d-8df8757737b9.png)

**Reply by lucasartsfr** [reply]
The project is online @rpearce if you wanna test : https://lucasarts.fr

**Reply by rpearce** [reply]
Thanks, @lucasartsfr. I haven't gotten back around to addressing this yet. I will use all the info you give me when I do!

**Reply by rpearce** [reply]
It looks like you're doing this to work around the issue:

```css
border-radius: calc(var(--BorderRadius) / .35);
```

I think a path to do this in the project might be something like...

1. Check if there's a border radius property on the `img`
2. Use the scaling information I have available at the time to maintain the perceived border radius

* * *

I've been doing other projects, but the issues / requests here are stacking up, so I need to dedicate some time back to this project.

**Reply by lucasartsfr** [reply]
Yep, the Fix look like this : Divide Radius by default scale (I do .35 because it was a test and close to all items, but, on the video, it's 0.414627 for one, 0.374583 on another etc..)

Yes, It could be a fix ! Or add a "BorderRadius" property to add Radius directly on the Element and not in CSS maybe. This allow you to juste Calculate it easily ?

**Reply by rpearce** [reply]
I forgot about this because I didn't make an issue. Here we go: https://github.com/rpearce/react-medium-image-zoom/issues/405

**Reply by rpearce** [reply]
I've got a simple, working solution, but this is one of those things I need to think about a little more, for people may not want to keep the border radius for the zoomed state.

It feels like it only makes sense for regular `<img />` elements and not `<div>`s with background images, nor `<img />` elements with `object-fit`.

API-wise, it feels like this may need to be a user-defined option 🤔 . I'll think on it some more, but let me know what you think if you can.

**Reply by lucasartsfr** [reply]
Hey ! I did not try with Div, just IMG so I don't know what could be the problem with div ?

**Reply by rpearce** [reply]
DIVs that use `background-size` and IMG elements that use `object-fit` have their content stretched or shrank within the element based on the values. If you have `cover` as an option, the image will be stretched to cover the element.

The way this library works with those is that it does maths to figure out the "actual" rendered dimensions of the image, and then it does its zooming based on that.

A good example: you have a source image that is 16:9, but your DIV is 4:3 (or dynamic and changes!). When you first see the DIV, parts of the image are cut off, but when you zoom, you get the entire, original image, and the animation for that just works--it looks like your entire photo is revealed, and then it moves to the center of the screen.

👆 with this scenario, the original shape is not maintained, for the image was previously constrained, and so the added border-radius makes it look a little different.

* * *

Wow that was a lot to type!

Maybe it doesn't matter, and people can choose? If an option is provided from a user, then it doesn't matter if it's an IMG or DIV--the person specified what they wanted.

**Reply by rpearce** [reply]
Are there things similar to `border-radius` that you can think of that someone might also want to be copied to the zoom image and scaled?

p.s. Here is it working for a regular image (I think)

https://user-images.githubusercontent.com/592876/234741263-48fff331-e438-47cd-9c93-8884aec892ea.mov

**Comment by lucasartsfr** [comment]
Hum.. nothing in my mind. I think we need to test to figure out what could
be affected by the scale effect

Le jeu. 27 avr. 2023 à 03:55, Robert Pearce ***@***.***> a
écrit :

**Reply by rpearce** [reply]
Ok. I'll think about a way to add this to the prop API and put a beta release out there within the next few days if I can. Thanks!

**Reply by rpearce** [reply]
@lucasartsfr can you try `react-medium-image-zoom@5.2.0-beta.1`? Here is the draft pull request for it: https://github.com/rpearce/react-medium-image-zoom/pull/417

In short, there is a new option, `zoomUseBorderRadius`, that would be used like

```js
<Zoom zoomUseBorderRadius={true}>
```

**Comment by lucasartsfr** [comment]
Hey Robert !

I'll Check that asap, I don't have many time for the moment, I am sorry,

Le ven. 28 avr. 2023 à 05:02, Robert Pearce ***@***.***> a
écrit :

**Reply by rpearce** [reply]
No problem, this will be here for you to evaluate when you're ready.

p.s. Your nature photos are beautiful!

**Comment by rpearce** [comment]
Given there's been no movement from the requester for some time, and I'm not 100% on the feature, I closed the issue for now (https://github.com/rpearce/react-medium-image-zoom/issues/405).

**Comment by lucasartsfr** [comment]
Hi ! I am sorry, I really don't have time for this at the moment ! Thank
you,

Le dim. 12 nov. 2023 à 15:31, Robert Pearce ***@***.***> a
écrit :
